src/gdb_main.c

```diff
@@ -841,53 +853,54 @@ static const cmd_executer_s v_commands[] = {
 	{NULL, NULL},
 };
 
-static void handle_v_packet(char *packet, const size_t plen)
+static void handle_v_packet(const gdb_packet_s *const packet)
 {
-	if (exec_command(packet, plen, v_commands))
+	if (exec_command(packet->data, packet->size, v_commands))
 		return;
 
+#ifndef DEBUG_GDB_IS_NOOP
 	/*
 	 * The vMustReplyEmpty is used as a feature test to check how gdbserver handles
 	 * unknown packets, don't print an error message for it.
 	 */
-	if (strcmp(packet, "vMustReplyEmpty") != 0)
-		DEBUG_GDB("*** Unsupported packet: %s\n", packet);
-	gdb_putpacketz("");
+	static const char *const must_reply_empty = "vMustReplyEmpty";
```

Comment:
Perhaps use `static const char []` here to leverage `ARRAY_LENGTH()` in the strncmp()?